rust/parquet/src/file/properties.rs

```diff
@@ -99,7 +99,7 @@ pub struct WriterProperties {
     max_row_group_size: usize,
     writer_version: WriterVersion,
     created_by: String,
-    key_value_metadata: Option<Vec<KeyValue>>,
+    pub(crate) key_value_metadata: Option<Vec<KeyValue>>,
```

Comment:
@sunchao this was the easiest way of making the metadata mutable without a lot of effort. I tried to create a `WriterPropertiesBuilder` from the `WriterProperties`, but for some properties there wasn't a way of retaining properties (esp if the user has specified compression/encoding options for some columns).

the mutability is required in order to encode the `ARROW:schema`. I followed the C++ approach, so that if a user writes Arrow data to Parquet, the schema information is retained when they read that file back in.

Reply:
+1. I think this is fine.